checks.d/mongo.py

```diff
@@ -304,19 +357,31 @@ def check(self, instance):
         except KeyError:
             pass
 
+        dbnames = cli.database_names()
+        for db_n in dbnames:
+            db_aux = cli[db_n]
+            dbstats[db_n] = {'stats': db_aux.command('dbstats')}
+
         # Go through the metrics and save the values
         for m in self.METRICS:
             # each metric is of the form: x.y.z with z optional
             # and can be found at status[x][y][z]
             value = status
-            try:
-                for c in m.split("."):
-                    value = value[c]
-            except KeyError:
+
+            if m.startswith('stats'):
                 continue
+            else:
+                try:
+                    for c in m.split("."):
+                        value = value[c]
+                except KeyError:
+                    continue
 
             # value is now status[x][y][z]
-            assert type(value) in (types.IntType, types.LongType, types.FloatType)
+            assert type(value) in (
+                types.IntType,
+                types.LongType,
+                types.FloatType)
```

Comment:
You should use something more pythonic and raise a more descriptive error. Something like:
```python
if not isinstance(value, (int, long, float)):
    raise TypeError('{0} value is {1}, it should be an int, a float or a long instead.'.format(m, type(value)))
```